### PR TITLE
EQL: case sensitivity aware integration testing

### LINKED ISSUE #56754: EQL: case sensitivity aware integration testing
With the [addition of `case_sensitivity` request parameter](https://github.com/elastic/elasticsearch/pull/56404), we should consider integration testing infra supporting it, as well, so that tests work no matter the value of this parameter.

### PR DESCRIPTION
**Pull request by astefan**
This PR is a wrap up of https://github.com/elastic/elasticsearch/pull/57323 and most of the code originates there.

Part of #57313
Closes #56754

Pulled over the latest TOML integration tests from Python, which have case (in)sensitive toggles.

There's a handful more tests added, and I noticed that some of the sequence tests are failing, so I left those in the exclusions list. Also updated the HLRC to include case sensitive toggles, which was needed for the tests.

Basically to mark a test as case-sensitive only, add this to the TOML
```
case_sensitive = true
```
or this to make a test case-insensitive only.
```
case_insensitive = true
```
If the flags aren't set, then the test is assumed to work with both and tests are run both with case_sensitive=true and case_sensitive=false.

### CODE REVIEW

File: x-pack/plugin/eql/qa/common/src/main/resources/test_queries.toml [hunk 1]
```diff
@@ -391,8 +691,12 @@ sequence
...
   [file where event_subtype_full == "file_delete_event"] by file_path
 | head 4
-| tail 2'''
-expected_event_ids  = [67, 68, 69, 70, 72, 73, 74, 75]
+| tail 2
+'''
+expected_event_ids  = [67, 68,
+                       69, 70,
+                       72, 73,
+                       74, 75]
```
Comment (costin) [inline x-pack/plugin/eql/qa/common/src/main/resources/test_queries.toml:699]: These are incorrectly formatted (as is the rest of the group under it) - the sequence has 4 items not 2 on a line.

**Review by costin — APPROVED** [review]
LGTM
